### DISCUSSION TITLE
Beta 11, how to handle progress, drawer ?

### DISCUSSION THREAD

**Discussion by michelson** [katspaugh/wavesurfer.js#2941] [Q&A] [answered]
Hi there, We have updated the beta of Wavesurfer, and in the previous version we did some calls to the backend and drawer objects, as they are not present in the current beta version what is the proper way to handle the following calls?

`wave.drawer.lastPos`
`wave.drawer.progress(percent)`
`wave.backend.getPlayedPercents()`

thanks

**Comment by katspaugh** [comment, marked as answer]
In v7, you have methods like `wavesurfer.getCurrentTime()`, `wavesurfer.setTime(seconds)`, `wavesurfer.seekTo(percentage)`.

`getPlayedPercents` would be `wavesurfer.getCurrentTime() / wavesurfer.getDuration()`.

**Reply by michelson** [reply]
Thanks! it works

**Comment by doctor30309** [comment]
Why does getDuraiton return "auto"?  It used to return the length of the wave file...

**Reply by katspaugh** [reply]
Example code and audio file?

**Reply by doctor30309** [reply]
The cursor that normally follows along with the play stopped working.  So I started digging.  I am also setting a label on the screen with the. duration of a wave file.  This stopped working.  So I am assume the two are tied together.

To see what the value is, I wrote a simple line of code to write to the log:
console.log("Setting Duration: " + wavesurfer.getDuration());

This writes "Setting Duration: auto" in the log...

**Reply by katspaugh** [reply]
It’s not what I asked, is it.

**Reply by doctor30309** [reply]
It won't let me upload the audio file as it is larger than 25MB.  Obviously, wav files don't really compress when zipped. So this attached wav is about 30 seconds of the actual wav file...

Here is the code I am using to test with...  

wavesurfer.on("audioprocess", function () { 
    $('[data-role="currentTime"]').text(formatTimecode(wavesurfer.getCurrentTime())); 
    $('[data-role="totalDuration"]').text(formatTimecode(wavesurfer.getDuration()));
});

[Call Down.wav.zip](https://github.com/katspaugh/wavesurfer.js/files/11885937/Call.Down.wav.zip)

**Reply by doctor30309** [reply]
Here is how I am initializing wavesurfer...


var div = document.createElement('div');
$(div).attr("id", "wave");
$(div).addClass("wave card-img-overlay");
$(div).data("role", "wave");
$('[data-role="wave-list"]').append(div);
var ctx = document.createElement('canvas').getContext('2d');
wavesurfer = WaveSurfer.create({
        container: $(div)[0],
        preload: "auto",
        waveColor: "#f5a637",
        progressColor: "#d44700",
        mediaControls: false,
        hideScrollbar: true,
        autoCenter: true,
        fillParent: true,
        minPxPerSec: 10,
        normalize: true,
        barWidth: 1,
        barRadius: 3,
        height: 150,
        plugins: [
            //WaveSurfer.playhead.create({ returnOnPause: true, moveOnSeek: true, draw: true }),
            //WaveSurfer.cursor.create({ showTime: true, opacity: 1, customShowTimeStyle: { 'background-color': '#000', color: '#fff', padding: '2px', 'font-size': '10px' } })
        ]
    });

**Reply by katspaugh** [reply]
The duration of the file you attached is reported as 29.5 seconds as expected.
You can drag-n-drop the original file here and see if it works for you: https://wavesurfer-js.org/examples/#pitch.js
Your code looks correct, so I think it's a problem with the audio file. Try re-encoding it to some other format.

**Reply by doctor30309** [reply]
I have 20 of these audio files that I recorded, mixed, and mastered in my studio.  They have a sample rate of 96kHz.  They are 24 bit wav files...  They play just fine on every platform including wavesurfer.

However, getDuration returns auto and the cursor does not follow the music.  All this started happening with the new library.  

Should I be using a different event other than audioprocess?  Should I do this on redraw?

**Reply by doctor30309** [reply]
Also, when I click on the waveform to advance the cursor to that position, I get this error:
Uncaught TypeError: Failed to set the 'currentTime' property on 'HTMLMediaElement': The provided double value is non-finite.

I am not coding for this.  I assume it is handled internal to the library..  

I am assuming all of this is related to the same issue.

**Reply by katspaugh** [reply]
Yes, it's definitely related because it needs to know the duration to calculate on what timestamp you clicked.
I can't really debug w/o having the file, so try uploading it to dropbox or something.

**Reply by doctor30309** [reply]
Ok..  So I trimmed my page down to the absolute basics and got rid of all the user interactions.  Hard coded a wav file to play when user clicks play button.  Everything works fine.  I am at a loss but will slowly start adding things back in to see what broke it...

**Reply by doctor30309** [reply]
This command is why the duration comes back "auto"...

wavesurfer.load(song, peakData, "auto");

I guess the logic has changed in new library....  One problem solved :)

**Reply by doctor30309** [reply]
Now getting this error on load....

Uncaught (in promise) TypeError: firstChannel.some is not a function

**Reply by doctor30309** [reply]
Here is the code.  Here is the wav file dropbox link:
https://www.dropbox.com/s/1pavpsq0xwiuiv3/1.%20Call%20Down%20the%20Thunder.wav?dl=0


ar wavesurfer;
var loopPlay = false;
var currentTrack = -1;
var continuousPlay = true;
const peakData = "[0.0001,-0.0002,0.0001,-0.0001,0.0001,-0.0002,0.0001,-0.0001,0.0001,-0.0002,0.0001,-0.0002,0.0002,-0.0001,0.0001,-0.0001,0.0001,-0.0001,0.0002,-0.0002,0.0001,-0.0001,0.0001,-0.0001,0.0321,-0.0371,0.038,-0.0209,0.061,-0.0425,0.0203,-0.0693,0.0469,-0.0558,0.0448,-0.0786,0.1295,-0.0549,0.0563,-0.1113,0.0787,-0.13,0.08,-0.0857,0.1363,-0.1761,0.1134,-0.1354,0.1625,-0.1052,0.1517,-0.1334,0.1753,-0.1512,0.089,-0.2041,0.1079,-0.1384,0.1111,-0.1438,0.1616,-0.0768,0.1561,-0.1145,0.128,-0.1884,0.046,-0.2213,0.1338,-0.1508,0.1008,-0.2246,0.0782,-0.2349,0.105,-0.0862,0.1467,-0.1038,0.2979,-0.1693,0.199,-0.1477,0.2043,-0.1938,0.3293,-0.1371,0.2758,-0.1692,0.306,-0.4248,0.2428,-0.1848,0.2782,-0.2237,0.282,-0.2693,0.2551,-0.1864,0.2956,-0.3978,0.2441,-0.6563,0.1544,-0.2306,0.2239,-0.2927,0.4106,-0.1967,0.176,-0.3672,0.4562,-0.1333,0.1525,-0.2972,0.3297,-0.2659,0.3922,-0.4237,0.3232,-0.3356,0.2942,-0.2305,0.3453,-0.2112,0.2036,-0.2907,0.334,-0.483,0.4236,-0.5456,0.2752,-0.1693,0.3391,-0.3947,0.3346,-0.3563,0.2566,-0.3053,0.4639,-0.2072,0.2319,-0.6807,0.4195,-0.2467,0.263,-0.3827,0.2402,-0.1859,0.3097,-0.5862,0.5568,-0.2312,0.2539,-0.2618,0.5906,-0.2822,0.3936,-0.3879,0.4226,-0.3671,0.2151,-0.3952,0.3388,-0.46,0.3738,-0.6635,0.3153,-0.5224,0.4504,-0.2029,0.2685,-0.6721,0.2339,-0.5766,0.3837,-0.3437,0.7488,-0.2971,0.9372,-0.6198,0.3692,-0.1932,0.2006,-0.3241,0.2762,-0.2958,0.2234,-0.1932,0.3079,-0.8066,0.4165,-0.3377,0.2727,-0.6054,0.2404,-0.237,0.2933,-0.3608,0.2487,-0.0938,0.7617,-0.0907,0.2643,-0.4031,0.2874,-0.4643,0.4071,-0.2191,0.2047,-0.4626,0.2646,-0.4683,0.2174,-0.6549,0.769,-0.5872,0.8306,-0.2721,0.3407,-0.3652,0.3456,-0.1426,0.7778,-0.3125,0.4841,-0.5507,0.4742,-0.1423,0.1992,-0.6991,0.513,-0.1325,0.3264,-0.1289,0.1278,-0.3013,0.2807,-0.2687,0.2974,-0.2234,0.7616,-0.3811,0.8235,-0.2654,0.1762,-0.7014,0.588,-0.3651,0.8145,-0.3605,0.2343,-0.35,0.3697,-0.6038,0.374,-0.728,0.4618,-0.1711,0.2573,-0.1894,0.1606,-0.4511,0.5323,-0.3456,0.5061,-0.1723,0.3344,-0.166,0.5273,-0.1345,0.5321,-0.23,0.3045,-0.19,0.462,-0.2719,0.2861,-0.3469,0.6559,-0.2918,0.5186,-0.74,0.5418,-0.485,0.2493,-0.3405,0.2326,-0.7695,0.3738,-0.3261,0.6174,-0.2379,0.3021,-0.564,0.2768,-0.36,0.2282,-0.3295,0.204,-0.3622,0.3557,-0.3114,0.2253,-0.5147,0.5594,-0.4687,0.3842,-0.9316,0.1672,-0.3071,0.1618,-0.1963,0.2921,-0.3519,0.4151,-0.7165,0.2447,-0.2185,0.1211,-0.4869,0.3974,-0.4959,0.6186,-0.3205,0.2286,-0.4321,0.3784,-0.8633,0.686,-0.6266,0.2073,-0.4075,0.2154,-0.3941,0.1493,-0.2869,0.3435,-0.3413,0.0813,-0.8488,0.3475,-0.4926,0.284,-0.1361,0.2357,-0.2395,0.2776,-0.2677,0.3821,-0.2265,0.4087,-0.1993,0.6534,-0.3038,0.5639,-0.168,0.602,-0.5533,0.406,-0.3764,0.3013,-0.2628,0.2491,-0.1784,0.2321,-0.2665,0.3213,-0.2742,0.2018,-0.6511,0.4459,-0.3685,0.1074,-0.6624,0.1553,-0.6137,0.26,-0.5259,0.3697,-0.2901,0.2502,-0.7942,0.3064,-0.4093,0.3377,-0.6652,0.7305,-0.1704,0.5525,-0.3461,0.6626,-0.3101,0.604,-0.3588,0.4461,-0.258,0.3594,-0.6721,0.2375,-0.3289,0.2396,-0.4833,0.1938,-0.2613,0.2478,-0.68,0.3447,-0.6745,0.2797,-0.3208,0.4602,-0.2085,0.2813,-0.288,0.4527,-0.6434,0.2828,-0.3131,0.3349,-0.5349,0.2235,-0.4264,0.3199,-0.4177,0.2901,-0.4646,0.517,-0.3554,0.2963,-0.6959,0.2679,-0.3292,0.3493,-0.3544,0.3688,-0.273,0.266,-0.7922,0.3477,-0.2395,0.3737,-0.3159,0.4044,-0.4108,0.4503,-0.1521,0.3607,-0.306,0.2834,-0.3179,0.3126,-0.5349,0.334,-0.3333,0.4815,-0.2335,0.5071,-0.3898,0.1538,-0.3516,0.4049,-0.3474,0.2799,-0.6558,0.9145,-0.1884,0.5895,-0.2505,0.1382,-0.6342,0.3537,-0.4601,0.4834,-0.4042,0.2914,-0.8951,0.4856,-0.3783,0.5237,-0.2171,0.4045,-0.4367,0.1306,-0.4481,0.197,-0.3698,0.6045,-0.372,0.1784,-0.2053,0.2594,-0.1885,0.3025,-0.5973,0.4448,-0.2304,0.2684,-0.3262,0.2347,-0.5037,0.4996,-0.2403,0.8134,-0.4503,0.4868,-0.2115,0.433,-0.2799,0.4834,-0.5276,0.3855,-0.5622,0.5326,-0.2547,0.367,-0.7662,0.3761,-0.4424,0.2689,-0.4542,0.4014,-0.3381,0.4644,-0.6858,0.2011,-0.5468,0.3783,-0.5279,0.4803,-0.2536,0.3606,-0.4939,0.2225,-0.4069,0.9299,-0.3688,0.1985,-0.7455,0.461,-0.3068,0.3057,-0.3898,0.2318,-0.4044,0.2086,-0.4234,0.606,-0.4803,0.6616,-0.4351,0.1924,-0.5549,0.3065,-0.165,0.3379,-0.7223,0.1583,-0.3161,0.2893,-0.3726,0.7478,-0.2803,0.1459,-0.3138,0.2579,-0.1648,0.317,-0.3833,0.7424,-0.415,0.1581,-0.4452,0.2357,-0.4237,0.435,-0.4483,0.1742,-0.2112,0.1317,-0.4831,0.392,-0.5545,0.5003,-0.2359,0.3823,-0.3769,0.7182,-0.2462,0.21,-0.5608,0.2125,-0.2334,0.189,-0.2845,0.3181,-0.3711,0.2869,-0.2225,0.3131,-0.3257,0.1595,-0.2937,0.4204,-0.1757,0.3026,-0.3282,0.3772,-0.2648,0.6798,-0.3386,0.5752,-0.6979,0.553,-0.325,0.5173,-0.4214,0.2212,-0.1033,0.1878,-0.4385,0.4177,-0.5395,0.2972,-0.5501,0.5298,-0.3134,0.4719,-0.2919,0.1209,-0.8873,0.8428,-0.7762,0.259,-0.2323,0.4326,-0.3089,0.3545,-0.5691,0.574,-0.3492,0.5009,-0.2447,0.8622,-0.5608,0.4121,-0.8273,0.483,-0.3743,0.4208,-0.4043,0.4976,-0.4058,0.4082,-0.5208,0.4529,-0.3123,0.1974,-0.3413,0.3329,-0.365,0.3382,-0.3216,0.2257,-0.6143,0.3775,-0.2023,0.1859,-0.4051,0.3537,-0.3227,0.3831,-0.1771,0.33,-0.4702,0.3702,-0.2437,0.2083,-0.5864,0.2193,-0.7023,0.2613,-0.3769,0.4267,-0.4548,0.2158,-0.2123,0.2639,-0.3034,0.359,-0.2298,0.3371,-0.7987,0.234,-0.2287,0.1775,-0.4176,0.5217,-0.3674,0.2501,-0.3598,0.1402,-0.4047,0.3978,-0.2889,0.5716,-0.2235,0.2482,-0.3271,0.2089,-0.6383,0.2761,-0.203,0.316,-0.2689,0.2462,-0.3622,0.289,-0.2747,0.3612,-0.4976,0.5255,-0.5967,0.8517,-0.599,0.2685,-0.3252,0.2177,-0.1873,0.356,-0.3111,0.4886,-0.2512,0.2021,-0.2415,0.5171,-0.7284,0.2395,-0.3349,0.3197,-0.1463,0.3602,-0.2998,0.4807,-0.2633,0.4438,-0.2409,0.4075,-0.578,0.2945,-0.4393,0.6649,-0.1411,0.447,-0.2536,0.2883,-0.16,0.3494,-0.8151,0.4687,-0.5948,0.3345,-0.3033,0.3943,-0.3842,0.2407,-0.3178,0.3186,-0.2798,0.7488,-0.4123,0.4391,-0.531,0.5858,-0.612,0.8978,-0.1524,0.2818,-0.4377,0.2243,-0.4926,0.4782,-0.2253,0.2362,-0.8626,0.7862,-0.3016,0.4816,-0.6901,0.1993,-0.9023,0.2412,-0.182,0.203,-0.2115,0.2164,-0.7552,0.19,-0.6096,0.5447,-0.3425,0.8408,-0.289,0.725,-0.2852,0.6119,-0.2534,0.6213,-0.3196,0.2863,-0.1743,0.2169,-0.3434,0.2487,-0.3107,0.2099,-0.2496,0.5709,-0.2751,0.6904,-0.2783,0.1422,-0.6813,0.5013,-0.1975,0.3272,-0.3992,0.2562,-0.3716,0.2484,-0.3321,0.3187,-0.4151,0.2879,-0.4807,0.3994,-0.8294,0.4106,-0.6699,0.6072,-0.427,0.509,-0.385,0.2573,-0.236,0.2192,-0.5875,0.2384,-0.2033,0.937,-0.2265,0.1848,-0.5811,0.1191,-0.7464,0.5663,-0.3212,0.3561,-0.3565,0.2735,-0.2359,0.2515,-0.3403,0.3002,-0.342,0.3474,-0.1738,0.1935,-0.5641,0.1978,-0.2903,0.7319,-0.2129,0.288,-0.156,0.4853,-0.5645,0.0001,-0.3639,0.0543,-0.5871,0.8098,-0.519,0.9172,-0.5232,0.3294,-0.3986,0.3325,-0.4409,0.2059,-0.4369,0.4351,-0.2506,0.5533,-0.2967,0.2482,-0.2029,0.2436,-0.5224,0.2645,-0.1591,0.2564,-0.3488,0.7351,-0.3858,0.4284,-0.469,0.3793,-0.278,0.2087,-0.3935,0.5488,-0.3928,0.4315,-0.7639,0.4414,-0.2702,0.3134,-0.8101,0.2963,-0.3425,0.2808,-0.3708,0.1842,-0.4113,0.2977,-0.3485,0.5793,-0.2028,0.5098,-0.272,0.8729,-0.3448,0.427,-0.3459,0.3151,-0.3509,0.3216,-0.2498,0.2024,-0.3205,0.2609,-0.5845,0.3513,-0.2048,0.2579,-0.6122,0.8743,-0.5512,0.3499,-0.7527,0.545,-0.2852,0.57,-0.1722,0.666,-0.0867,0.5436,-0.4384,0.2032,-0.3523,0.7686,-0.5614,0.3901,-0.6718,0.6152,-0.5045,0.2525,-0.2618,0.7481,-0.376,0.2319,-0.3647,0.4084,-0.5774,0.3094,-0.1726,0.2798,-0.6326,0.2583,-0.3446,0.5117,-0.6988,0.2442,-0.4587,0.1971,-0.2948,0.5535,-0.5449,0.9132,-0.4796,0.234,-0.4767,0.7337,-0.5637,0.2082,-0.4318,0.2619,-0.276,0.386,-0.7909,0.6566,-0.5205,0.267,-0.4169,0.2961,-0.3641,0.4092,-0.3294,0.6841,-0.4394,0.6629,-0.0652,0.2144,-0.3635,0.4763,-0.1749,0.3354,-0.143,0.196,-0.3063,0.359,-0.2716,0.6707,-0.292,0.8088,-0.4247,0.4107,-0.6734,0.1717,-0.7498,0.7344,-0.4105,0.4627,-0.2754,0.2983,-0.3808,0.4036,-0.5278,0.34,-0.311,0.139,-0.4485,0.3176,-0.6727,0.2805,-0.6335,0.3773,-0.4194,0.4485,-0.3083,0.3424,-0.9121,0.4292,-0.4247,0.5214,-0.6441,0.5064,-0.6231,0.4745,-0.3801,0.0942,-0.5331,0.8368,-0.353,0.4788,-0.4068,0.4104,-0.8127,0.4261,-0.4168,0.2321,-0.2584,0.8524,-0.3869,0.8437,-0.3921,0.6363,-0.6915,0.6668,-0.255,0.9164,-0.3162,0.2633,-0.2941,0.2344,-0.3347,0.2493,-0.4364,0.6455,-0.3454,0.4864,-0.2224,0.1462,-0.3602,0.3024,-0.2208,0.1958,-0.3478,0.3032,-0.4199,0.2613,-0.2341,0.3441,-0.4363,0.8789,-0.2283,0.3673,-0.9032,0.618,-0.4346,0.4723,-0.2882,0.4083,-0.5888,0.2656,-0.4769,0.3033,-0.4229,0.2164,-0.3096,0.3895,-0.3094,0.2237,-0.3026,0.4705,-0.346,0.3393,-0.5838,0.3213,-0.3524,0.5394,-0.2116,0.213,-0.2699,0.564,-0.6302,0.5195,-0.7079,0.2792,-0.6782,0.2624,-0.3386,0.3613,-0.3883,0.5309,-0.2204,0.1797,-0.3895,0.7863,-0.6015,0.4148,-0.3187,0.6265,-0.6699,0.8848,-0.4026,0.4915,-0.5235,0.316,-0.3589,0.4242,-0.5544,0.787,-0.5594,0.4094,-0.2757,0.4793,-0.5356,0.2811,-0.2267,0.2827,-0.2339,0.4699,-0.2468,0.4107,-0.4767,0.3798,-0.2772,0.4496,-0.136,0.2391,-0.2666,0.1261,-0.2363,0.3483,-0.8426,0.3848,-0.4458,0.233,-0.4065,0.3914,-0.128,0.4725,-0.2713,0.412,-0.9056,0.3089,-0.3592,0.3744,-0.2501,0.5028,-0.4581,0.5588,-0.537,0.2462,-0.5485,0.629,-0.5517,0.2896,-0.5979,0.3272,-0.5858,0.3038,-0.6041,0.6213,-0.2257,0.3795,-0.346,0.577,-0.2623,0.388,-0.2703,0.3846,-0.1508,0.3874,-0.596,0.4328,-0.2674,0.3726,-0.4337,0.5923,-0.4524,0.615,-0.3637,0.4381,-0.348,0.5581,-0.7661,0.5151,-0.5918,0.6953,-0.7288,0.3898,-0.4061,0.5338,-0.406,0.598,-0.5266,0.4789,-0.802,0.5898,-0.5236,0.4197,-0.3413,0.5514,-0.3967,0.3199,-0.4265,0.2867,-0.5032,0.7289,-0.5888,0.275,-0.3863,0.3765,-0.5127,0.2281,-0.2759,0.1086,-0.8193,0.5374,-0.2405,0.2128,-0.1805,0.3473,-0.195,0.2201,-0.4717,0.173,-0.2943,0.6678,-0.2073,0.2922,-0.2276,0.3468,-0.5473,0.5881,-0.3584,0.2906,-0.5231,0.2035,-0.2359,0.2737,-0.3096,0.3481,-0.2375,0.7049,-0.214,0.2385,-0.4858,0.3156,-0.9231,0.0495,-0.2998,0.4145,-0.2857,0.2568,-0.281,0.2529,-0.3342,0.3931,-0.1296,0.2668,-0.5666,0.6717,-0.1095,0.3253,-0.2691,0.6244,-0.3961,0.367,-0.2461,0.3996,-0.2811,0.2493,-0.1795,0.1812,-0.2208,0.4629,-0.2114,0.1381,-0.1976,0.3063,-0.4307,0.5368,-0.4329,0.2303,-0.2531,0.246,-0.4383,0.3573,-0.3794,0.283,-0.2843,0.3142,-0.2328,0.502,-0.4242,0.3196,-0.5767,0.2086,-0.7531,0.8563,-0.34,0.5372,-0.5369,0.3616,-0.2429,0.5466,-0.4026,0.7323,-0.2485,0.3896,-0.2671,0.2306,-0.7435,0.3404,-0.3102,0.2205,-0.3288,0.4749,-0.1923,0.3943,-0.3279,0.3887,-0.2377,0.6332,-0.269,0.2829,-0.8576,0.423,-0.491,0.6333,-0.2106,0.3719,-0.0817,0.4705,-0.2102,0.2095,-0.3873,0.3897,-0.4135,0.5083,-0.5834,0.224,-0.2918,0.3823,-0.5626,0.4005,-0.5144,0.3753,-0.4242,0.2997,-0.4739,0.1602,-0.4195,0.4311,-0.1351,0.4304,-0.398,0.3031,-0.4298,0.5746,-0.3984,0.6902,-0.2566,0.4442,-0.2502,0.3957,-0.2855,0.1812,-0.2539,0.2434,-0.4659,0.5549,-0.2698,0.4293,-0.2788,0.4926,-0.6662,0.5803,-0.4517,0.6691,-0.1918,0.5718,-0.4598,0.2516,-0.5303,0.4978,-0.3062,0.5336,-0.3607,0.8922,-0.4436,0.1243,-0.6478,0.2924,-0.2103,0.3432,-0.1455,0.4369,-0.1951,0.3969,-0.6315,0.5337,-0.2902,0.7443,-0.2807,0.6554,-0.5676,0.0353,-0.8963,0.1951,-0.6513,0.4675,-0.3509,0.4914,-0.4334,0.311,-0.8061,0.4024,-0.2399,0.2163,-0.152,0.198,-0.481,0.2579,-0.2114,0.5642,-0.2125,0.3682,-0.342,0.2327,-0.7288,0.7031,-0.2357,0.5458,-0.8296,0.2282,-0.2963,0.7092,-0.1811,0.3219,-0.7339,0.6174,-0.4386,0.6175,-0.3404,0.2404,-0.607,0.3159,-0.6112,0.4618,-0.3644,0.4961,-0.2017,0.6737,-0.1195,0.1566,-0.5854,0.3506,-0.552,0.3827,-0.4165,0.312,-0.4519,0.3463,-0.2946,0.555,-0.4343,0.3058,-0.1937,0.236,-0.2411,0.0941,-0.4036,0.2273,-0.2439,0.357,-0.2793,0.3273,-0.6945,0.3521,-0.5726,0.4596,-0.4707,0.835,-0.2931,0.6071,-0.7994,0.592,-0.7272,0.5044,-0.5711,0.5559,-0.4171,0.1454,-0.2441,0.6151,-0.2854,0.2228,-0.2788,0.5064,-0.5443,0.7188,-0.1294,0.38,-0.6699,0.6798,-0.1978,0.5567,-0.3789,0.2031,-0.2583,0.3597,-0.4997,0.3882,-0.4486,0.7387,-0.3882,0.6274,-0.4096,0.3349,-0.3211,0.3688,-0.3444,0.1235,-0.2316,0.2652,-0.6735,0.3631,-0.3391,0.2347,-0.6099,0.6088,-0.1617,0.7696,-0.3926,0.6605,-0.7407,0.2057,-0.1925,0.4544,-0.4956,0.1264,-0.5121,0.5045,-0.3199,0.3427,-0.4498,0.378,-0.9005,0.4819,-0.3758,0.1448,-0.813,0.3588,-0.2948,0.2955,-0.6096,0.2153,-0.256,0.4086,-0.3654,0.275,-0.185,0.4893,-0.407,0.4771,-0.2604,0.6813,-0.3761,0.802,-0.1998,0.7365,-0.4744,0.1105,-0.2559,0.3417,-0.4558,0.2493,-0.5583,0.3492,-0.4995,0.5116,-0.3733,0.5262,-0.9271,0.4345,-0.2545,0.3409,-0.2815,0.5489,-0.5375,0.4288,-0.284,0.3969,-0.6296,0.4779,-0.3793,0.389,-0.3644,0.4253,-0.2132,0.2785,-0.3794,0.2559,-0.2974,0.2464,-0.3321,0.353,-0.2839,0.3521,-0.2539,0.5573,-0.6406,0.2945,-0.2163,0.6648,-0.3299,0.3192,-0.2779,0.8653,-0.392,0.3876,-0.2906,0.2175,-0.6118,0.3191,-0.4729,0.4087,-0.276,0.5337,-0.2819,0.4773,-0.3864,0.6447,-0.2502,0.5033,-0.6056,0.3566,-0.2144,0.5316,-0.454,0.6594,-0.177,0.5443,-0.2662,0.2713,-0.2387,0.3555,-0.3166,0.6865,-0.3373,0.4095,-0.3765,0.4838,-0.5533,0.3582,-0.3944,0.4905,-0.2294,0.6135,-0.5994,0.593,-0.6455,0.4219,-0.7246,0.445,-0.6694,0.6399,-0.5713,0.3304,-0.608,0.1827,-0.599,0.4918,-0.2577,0.2128,-0.8588,0.7736,-0.3398,0.2327,-0.8129,0.2334,-0.451,0.6971,-0.2357,0.2638,-0.5839,0.2679,-0.2548,0.3983,-0.2174,0.3209,-0.5042,0.7895,-0.468,0.5987,-0.4587,0.3979,-0.477,0.3504,-0.2487,0.2537,-0.4943,0.693,-0.3046,0.5269,-0.3186,0.4943,-0.5805,0.5401,-0.317,0.4552,-0.8071,0.7919,-0.6693,0.3473,-0.2366,0.5233,-0.6915,0.2799,-0.3556,0.5865,-0.2009,0.5801,-0.6088,0.5127,-0.9004,0.3848,-0.4227,0.373,-0.2641,0.4578,-0.4325,0.2449,-0.3377,0.7467,-0.3991,0.311,-0.5044,0.457,-0.6714,0.3752,-0.4655,0.3558,-0.8781,0.5706,-0.3371,0.307,-0.2146,0.1853,-0.3823,0.4455,-0.5799,0.245,-0.2631,0.5306,-0.3637,0.5474,-0.7881,0.2948,-0.8631,0.7312,-0.4704,0.5123,-0.3735,0.1633,-0.7135,0.3272,-0.8974,0.3933,-0.4619,0.4576,-0.4347,0.4775,-0.2219,0.1949,-0.2744,0.5313,-0.2125,0.5634,-0.3969,0.4997,-0.3415,0.5064,-0.3187,0.2841,-0.4486,0.4074,-0.6027,0.3112,-0.5532,0.3833,-0.7351,0.31,-0.5356,0.3514,-0.3576,0.4666,-0.6245,0.5466,-0.4717,0.649,-0.3678,0.3708,-0.2131,0.441,-0.5118,0.414,-0.4175,0.5131,-0.3326,0.582,-0.3244,0.3835,-0.6307,0.5637,-0.3616,0.2199,-0.6497,0.4289,-0.1499,0.3045,-0.5226,0.2595,-0.3683,0.7484,-0.5169,0.4956,-0.4123,0.5343,-0.7018,0.5798,-0.3479,0.4491,-0.3753,0.6479,-0.397,0.2412,-0.8398,0.7219,-0.791,0.6124,-0.5979,0.4304,-0.6388,0.5418,-0.533,0.2768,-0.4781,0.4153,-0.5735,0.6341,-0.716,0.4086,-0.6556,0.4677,-0.7273,0.2825,-0.5218,0.6694,-0.5385,0.6011,-0.4611,0.5594,-0.4316,0.8924,-0.4558,0.4399,-0.8812,0.6381,-0.5366,0.3001,-0.9379,0.5132,-0.5066,0.4733,-0.3096,0.3393,-0.3243,0.332,-0.3545,0.1373,-0.7326,0.527,-0.6988,0.5256,-0.3665,0.7334,-0.6927,0.45,-0.851,0.4325,-0.4798,0.5476,-0.3616,0.2955,-0.5792,0.3223,-0.3652,0.4103,-0.5186,0.5401,-0.3446,0.4645,-0.4558,0.2956,-0.3607,0.2307,-0.1788,0.1195,-0.11,0.066,-0.1167,0.1536,-0.1095,0.163,-0.1054,0.1548,-0.1036,0.0347,-0.0646,0.0363,-0.0546,0.0574,-0.026,0.0261,-0.021,0.0292,-0.0627,0.035,-0.0234,0.0695,-0.0385,0.0385,-0.0702,0.0055,-0.0329,0.0138,-0.0473,0.024,-0.008,0.0133,-0.0073,0.0079,-0.0119,0.0329,-0.0059,0.0316,-0.0049,0.024,-0.0054,0.013,-0.0184,0.0183,-0.008,0.0057,-0.004,0.0057,-0.0036,0.0026,-0.004,0.0052,-0.0026,0.0102,-0.0023,0.0043,-0.0031,0.0021,-0.0025,0.0017,-0.002,0.0016,-0.0018,0.001,-0.003,0.0009,-0.0007,0.0013,-0.001,0.0011,-0.0006,0.0009,-0.0004,0.0001,-0.0011,0.0008,-0.0004,0.0001,-0.0002,0.0001,-0.0002,0,0,0,0]";
const song = "/music/midnight outlaws/no warning/call down the thunder.wav";
const duration = 388000;
function initializeWaveSurfer() {
    console.log("Initializing Waversurfer");
    $("#wave-form-placeholder").addClass("visually-hidden");
    var div = document.createElement('div');
    $(div).attr("id", "wave");
    $(div).addClass("wave card-img-overlay");
    $(div).data("role", "wave");
    $('[data-role="wave-list"]').append(div);
    var ctx = document.createElement('canvas').getContext('2d');
    wavesurfer = WaveSurfer.create({
        container: $(div)[0],
        preload: "auto",
        waveColor: "#f5a637",
        progressColor: "#d44700",
        mediaControls: false,
        hideScrollbar: true,
        autoCenter: true,
        fillParent: true,
        minPxPerSec: 10,
        normalize: true,
        barWidth: 1,
        barRadius: 3,
        height: 150,
        peaks: peakData,
        url: song,
        duration: duration,
        plugins: []
    });
    wavesurfer.once('ready', function () { console.log('Ready Once Event Fired'); });
    wavesurfer.on('error', function (e) { console.warn(e); });
    wavesurfer.on('play', function (e) { console.log("Play Event Fired"); });
    wavesurfer.on('pause', function (e) { console.log("Pause Event Fired"); });
    wavesurfer.on('mute', function (e) { console.log("Mute Event Fired: " + wavesurfer.getMuted()); });
    wavesurfer.on('loading', function (p) { console.log("Loading Event Fired"); });
    wavesurfer.on('ready', function () { console.log("Ready Event Fired"); });
    wavesurfer.on('redraw', function () { console.log("Redraw Event Fired"); });
    wavesurfer.on("audioprocess", function () { console.log("Audioprocess Event Fired"); });
    wavesurfer.on("finish", function () { console.log("Finish Event Fired"); });

    console.log("Waversurfer is initialized...");
};

/************************************************************/
/*                      Initialize UI                       */
/************************************************************/
document.addEventListener('DOMContentLoaded', function () {
    initializeWaveSurfer();
    //wavesurfer.load(song, peakData, "auto");
    //wavesurfer.load(song);
    $('[data-action="play"]').on('click', function () { wavesurfer.playPause(); });
});

**Reply by doctor30309** [reply]
This error is generated when loading the wave file as part of the initialize routine..

Uncaught (in promise) TypeError: firstChannel.some is not a function
    at normalize (decoder.js:11:22)
    at Object.createBuffer (decoder.js:33:5)
    at WaveSurfer.load (wavesurfer.js:159:40)
    at new WaveSurfer (wavesurfer.js:42:18)
    at WaveSurfer.create (wavesurfer.js:20:16)
    at initializeWaveSurfer (mediaPlayer.js?v=ANARBG_c_-LsCtxiyc0r7iXvmD4_FStw1muMA79mheA:20:29)
    at HTMLDocument.<anonymous> (mediaPlayer.js?v=ANARBG_c_-LsCtxiyc0r7iXvmD4_FStw1muMA79mheA:57:5)

**Reply by doctor30309** [reply]
If I changed the initialize to not provide the peakdata, then I get no error...

**Reply by doctor30309** [reply]
Sorry to be a PIA.  Fixed the peak load problem...  Works fine...

**Reply by doctor30309** [reply]
Scratch everything.  I have it working except now the waveform does not fit in view and scrolls which I don't want.
